query/README.md

```diff
@@ -1,66 +1,203 @@
 # [MLIR](https://mlir.llvm.org) + [hail](https://hail.is) = 🚀🧬?
 
-## Dependencies
+## Installing build tools
 
-You will need `cmake` and `ninja`. If you're on OS X, use https://brew.sh:
+You will need to install `cmake` and `ninja`. If you're on macOS, you can do so
+using [Homebrew](https://brew.sh):
 
 ```sh
 brew install cmake ninja
 ```
 
-## Building/Installing LLVM and MLIR
-Obviously, update the paths for your environment. `$HAIL_DIR` is the root of the
-hail repository.
+## Setting up your shell environment
+
+In your shell configuration file (e.g. `~/.bashrc` for `bash`), add the following,
+and update variable values where specified:
+
+```sh
+# replace with the path to your local clone of the Hail repository
+export HAIL_DIRECTORY=$HOME/hail
+export HAIL_DEVBIN_DIRECTORY=$HAIL_DIRECTORY/devbin
+export HAIL_NATIVE_COMPILER_DIRECTORY=$HAIL_DIRECTORY/query
+export HAIL_NATIVE_COMPILER_BUILD_DIRECTORY=$HAIL_NATIVE_COMPILER_DIRECTORY/build
+
+# replace with the path to which you would like to clone the LLVM repository
+export LLVM_DIRECTORY=$HOME/llvm-project
+export LLVM_BUILD_DIRECTORY=$LLVM_DIRECTORY/build
+export LLVM_BUILD_BIN_DIRECTORY=$LLVM_BUILD_DIRECTORY/bin
+
+export PATH=$LLVM_BUILD_BIN_DIRECTORY:$HAIL_DEVBIN_DIRECTORY:$PATH
+```
+
+Refresh your shell configuration. For example, if you use `bash`, run:
+
+```sh
+source ~/.bashrc
+```
+
+## Building LLVM
+
+Clone the LLVM repository:
+
+```sh
+git clone https://github.com/llvm/llvm-project.git $LLVM_DIRECTORY
+```
+
+Edit the `cmake` invocation in `$HAIL_DEVBIN_DIRECTORY/hail-build-llvm`,
+uncommenting the optional flags as desired.
```

Comment:
editing devbin is unpleasant because it shows up as a diff in git. Maybe we should have `HAIL_QUERY_CMAKE_EXTRA_ARGS` as an envvar people set in their zshrc.

Reply:
Added optional arguments to the script, which feels more ergonomic to me than more environment variables 😸